src/lib/ptree/ptree.lua

```diff
@@ -283,19 +284,34 @@ function Manager:remove_stale_workers()
    end
 end
 
-function Manager:acquire_cpu_for_worker(id, app_graph)
+local function collect_pci_addresses (configuration)
+   local function is_device (pciaddr)
+      if lib.is_iface(pciaddr) then return false end
+      return pci.exists(pciaddr)
+   end
+   if configuration.softwire_config == nil then return {} end
    local pci_addresses = {}
-   -- Grovel through app initargs for keys named "pciaddr".  Hacky!
-   for name, init in pairs(app_graph.apps) do
-      if type(init.arg) == 'table' then
-         for k, v in pairs(init.arg) do
-            if k == 'pciaddr' and not lib.is_iface(v) then
-               table.insert(pci_addresses, v)
+   local instances = configuration.softwire_config.instance
+   for pciaddr,instance in pairs(instances) do
+      if is_device(pciaddr) then
+         pci_addresses[pciaddr] = true
+      end
+      for _,queue in ipairs(instance.queue.values) do
+         for k,v in pairs(queue.external_interface) do
```

Comment:
It's probably not appropriate to reference lwaftr-specific data paths (`queue`) from the generic ptree code.  What bug is being fixed here? :)